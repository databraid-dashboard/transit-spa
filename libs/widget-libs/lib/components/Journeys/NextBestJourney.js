var _jsxFileName = '../../src/components/Journeys/NextBestJourney.jsx',
    _this = this;

import React from 'react';
import PropTypes from 'prop-types';

import { List, Segment } from 'semantic-ui-react';

import TimeToLeave from './Journey/TimeToLeave';
import JourneyVisualization from './Journey/JourneyVisualization';
import ArriveByEstimate from './Journey/ArriveByEstimate';
import CurrentConditionsStatus from './Journey/CurrentConditionsStatus';

var NextBestJourney = function NextBestJourney(_ref) {
  var timeToLeaveInSeconds = _ref.timeToLeaveInSeconds,
      steps = _ref.steps,
      eta = _ref.eta,
<<<<<<< HEAD
      conditionStatus = _ref.conditionStatus,
      callRefreshJourneys = _ref.callRefreshJourneys;
=======
      conditionStatus = _ref.conditionStatus;
>>>>>>> Inital dashboard integration
  return React.createElement(
    Segment,
    { tertiary: true, __source: {
        fileName: _jsxFileName,
<<<<<<< HEAD
        lineNumber: 18
=======
        lineNumber: 12
>>>>>>> Inital dashboard integration
      },
      __self: _this
    },
    React.createElement(
      List,
      { divided: true, horizontal: true, size: 'huge', __source: {
          fileName: _jsxFileName,
<<<<<<< HEAD
          lineNumber: 19
        },
        __self: _this
      },
      React.createElement(TimeToLeave, {
        timeToLeaveInSeconds: timeToLeaveInSeconds,
        callRefreshJourneys: callRefreshJourneys,
        __source: {
          fileName: _jsxFileName,
          lineNumber: 20
=======
          lineNumber: 13
        },
        __self: _this
      },
      React.createElement(TimeToLeave, { timeToLeaveInSeconds: timeToLeaveInSeconds, __source: {
          fileName: _jsxFileName,
          lineNumber: 14
>>>>>>> Inital dashboard integration
        },
        __self: _this
      }),
      React.createElement(JourneyVisualization, { active: true, steps: steps, __source: {
          fileName: _jsxFileName,
<<<<<<< HEAD
          lineNumber: 24
=======
          lineNumber: 15
>>>>>>> Inital dashboard integration
        },
        __self: _this
      }),
      React.createElement(ArriveByEstimate, { eta: eta, __source: {
          fileName: _jsxFileName,
<<<<<<< HEAD
          lineNumber: 25
=======
          lineNumber: 16
>>>>>>> Inital dashboard integration
        },
        __self: _this
      }),
      React.createElement(CurrentConditionsStatus, { conditionStatus: conditionStatus, __source: {
          fileName: _jsxFileName,
<<<<<<< HEAD
          lineNumber: 26
=======
          lineNumber: 17
>>>>>>> Inital dashboard integration
        },
        __self: _this
      })
    )
  );
};

NextBestJourney.propTypes = {
  timeToLeaveInSeconds: PropTypes.number.isRequired,
  steps: PropTypes.arrayOf(PropTypes.object).isRequired,
  eta: PropTypes.string.isRequired,
<<<<<<< HEAD
  conditionStatus: PropTypes.string.isRequired,
  callRefreshJourneys: PropTypes.func.isRequired
=======
  conditionStatus: PropTypes.string.isRequired
>>>>>>> Inital dashboard integration
};

NextBestJourney.defaultProps = {
  timeToLeaveInSeconds: 1,
  steps: [{}],
  eta: '',
<<<<<<< HEAD
  conditionStatus: 'on-time',
  callRefreshJourneys: function callRefreshJourneys() {}
=======
  conditionStatus: 'on-time'
>>>>>>> Inital dashboard integration
};

export default NextBestJourney;