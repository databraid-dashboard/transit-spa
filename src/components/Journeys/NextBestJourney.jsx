import React from 'react';
import PropTypes from 'prop-types';

import { List, Segment } from 'semantic-ui-react';

import TimeToLeave from './Journey/TimeToLeave';
import JourneyVisualization from './Journey/JourneyVisualization';
import ArriveByEstimate from './Journey/ArriveByEstimate';
import CurrentConditionsStatus from './Journey/CurrentConditionsStatus';

const NextBestJourney = ({
  timeToLeaveInSeconds,
  steps,
  eta,
  conditionStatus,
  callRefreshJourneys,
}) =>
  (<Segment tertiary>
    <List horizontal size="huge">
      <TimeToLeave
        timeToLeaveInSeconds={timeToLeaveInSeconds}
        callRefreshJourneys={callRefreshJourneys}
      />
      <JourneyVisualization active steps={steps} />
      <ArriveByEstimate eta={eta} />
      <CurrentConditionsStatus conditionStatus={conditionStatus} />
    </List>
  </Segment>);

NextBestJourney.propTypes = {
  timeToLeaveInSeconds: PropTypes.number.isRequired,
  steps: PropTypes.arrayOf(PropTypes.object).isRequired,
  eta: PropTypes.string.isRequired,
  conditionStatus: PropTypes.string.isRequired,
  callRefreshJourneys: PropTypes.func.isRequired,
};

NextBestJourney.defaultProps = {
  timeToLeaveInSeconds: 1,
  steps: [{}],
  eta: '',
  conditionStatus: 'on-time',
  callRefreshJourneys: () => {},
};

export default NextBestJourney;
