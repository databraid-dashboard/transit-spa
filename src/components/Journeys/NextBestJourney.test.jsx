import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import NextBestJourney from './NextBestJourney';

describe('NextBestJourney', () => {
  it('should render when passed required props', () => {
    const timeToLeaveInSeconds = 500;
    const steps = [
      { duration: '5 mins', instruction: 'Walk to Some Other St. Station', mode: 'WALKING' },
    ];
    const eta = '12:00pm';
    const conditionStatus = 'on-time';

    const component = mount(
      <NextBestJourney
        timeToLeaveInSeconds={timeToLeaveInSeconds}
        steps={steps}
        eta={eta}
        conditionStatus={conditionStatus}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
