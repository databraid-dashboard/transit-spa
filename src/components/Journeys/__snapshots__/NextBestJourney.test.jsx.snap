// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextBestJourney should render when passed required props 1`] = `
<NextBestJourney
  callRefreshJourneys={[Function]}
  conditionStatus="on-time"
  eta="12:00pm"
  steps={
    Array [
      Object {
        "duration": "5 mins",
        "instruction": "Walk to Some Other St. Station",
        "mode": "WALKING",
      },
    ]
  }
  timeToLeaveInSeconds={500}
>
  <Segment
    tertiary={true}
  >
    <div
      className="ui tertiary segment"
    >
      <List
        divided={true}
        horizontal={true}
        size="huge"
      >
        <div
          className="ui huge divided horizontal list"
          role="list"
        >
          <TimeToLeave
            callRefreshJourneys={[Function]}
            timeToLeaveInSeconds={500}
          >
            <ListItem>
              <div
                className="item"
                onClick={[Function]}
                role="listitem"
              >
                <Icon
                  as="i"
                  name="hourglass half"
                  size="large"
                >
                  <i
                    aria-hidden="true"
                    className="hourglass half large icon"
                  />
                </Icon>
                <ListContent>
                  <div
                    className="content"
                  >
                    <ListHeader>
                      <div
                        className="header"
                      >
                        leave in
                      </div>
                    </ListHeader>
                    <Timer
                      onComplete={[Function]}
                      seconds={500}
                    >
                      <div>
                        08
                        :
                        20
                      </div>
                    </Timer>
                  </div>
                </ListContent>
              </div>
            </ListItem>
          </TimeToLeave>
          <JourneyVisualization
            active={true}
            steps={
              Array [
                Object {
                  "duration": "5 mins",
                  "instruction": "Walk to Some Other St. Station",
                  "mode": "WALKING",
                },
              ]
            }
          >
            <ListItem>
              <div
                className="item"
                onClick={[Function]}
                role="listitem"
              >
                <StepGroup>
                  <div
                    className="ui steps"
                  >
                    <Step
                      disabled={true}
                    >
                      <div
                        className="disabled step"
                        onClick={[Function]}
                      >
                        <Icon
                          as="i"
                          name="blind"
                          size="big"
                        >
                          <i
                            aria-hidden="true"
                            className="blind big icon"
                          />
                        </Icon>
                        <StepContent
                          title="Walk"
                        >
                          <div
                            className="content"
                          >
                            <StepTitle
                              title="Walk"
                            >
                              <div
                                className="title"
                              >
                                Walk
                              </div>
                            </StepTitle>
                          </div>
                        </StepContent>
                      </div>
                    </Step>
                  </div>
                </StepGroup>
              </div>
            </ListItem>
          </JourneyVisualization>
          <ArriveByEstimate
            eta="12:00pm"
          >
            <ListItem>
              <div
                className="item"
                onClick={[Function]}
                role="listitem"
              >
                <Icon
                  as="i"
                  name="clock"
                  size="large"
                >
                  <i
                    aria-hidden="true"
                    className="clock large icon"
                  />
                </Icon>
                <ListContent>
                  <div
                    className="content"
                  >
                    <ListHeader>
                      <div
                        className="header"
                      >
                        ETA
                      </div>
                    </ListHeader>
                    12:00pm
                  </div>
                </ListContent>
              </div>
            </ListItem>
          </ArriveByEstimate>
          <CurrentConditionsStatus
            conditionStatus="on-time"
          >
            <ListItem>
              <div
                className="item"
                onClick={[Function]}
                role="listitem"
              >
                <ListContent>
                  <div
                    className="content"
                  >
                    <ListHeader>
                      <div
                        className="header"
                      >
                        status:
                      </div>
                    </ListHeader>
                    <Popup
                      content="on-time"
                      inverted={true}
                      offset={50}
                      on="hover"
                      position="right center"
                      style={
                        Object {
                          "borderRadius": 0,
                          "opacity": 0.7,
                          "padding": "2em",
                        }
                      }
                      trigger={
                        <Icon
                          as="i"
                          name="checkmark"
                          size="large"
                        />
                      }
                    >
                      <Portal
                        closeOnDocumentClick={true}
                        closeOnEscape={true}
                        closeOnTriggerMouseLeave={true}
                        mouseEnterDelay={50}
                        mouseLeaveDelay={70}
                        onClose={[Function]}
                        onMount={[Function]}
                        onOpen={[Function]}
                        onUnmount={[Function]}
                        openOnTriggerClick={true}
                        openOnTriggerMouseEnter={true}
                        trigger={
                          <Icon
                            as="i"
                            name="checkmark"
                            size="large"
                          />
                        }
                      >
                        <Icon
                          as="i"
                          name="checkmark"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          size="large"
                        >
                          <i
                            aria-hidden="true"
                            className="checkmark large icon"
                            onBlur={[Function]}
                            onClick={[Function]}
                            onFocus={[Function]}
                            onMouseEnter={[Function]}
                            onMouseLeave={[Function]}
                          />
                        </Icon>
                      </Portal>
                    </Popup>
                  </div>
                </ListContent>
              </div>
            </ListItem>
          </CurrentConditionsStatus>
        </div>
      </List>
    </div>
  </Segment>
</NextBestJourney>
`;
